File: snippets.py
# Note: Use of CollectionRef stream() is prefered to get()
docs = (
    db.collection("cities")
    .where(filter=FieldFilter("capital", "==", True))
    .stream()
)

async for doc in docs:
    print(f"{doc.id} => {doc.to_dict()}")

File: snippets.cpp
auto db = firestore::Firestore::Create(app).get();

auto result = db.Collection("cities")
                    .Where("capital", "==", true)
                    .Get()
                    .get();

for (const auto& doc : result.documents()) {
  std::cout << doc.id() << " => " << doc.GetData() << std::endl;
}
